flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/Mail.java

```diff
@@ -61,11 +62,7 @@ public String toString() {
 		return String.format(descriptionFormat, descriptionArgs);
 	}
 
-	public void run() {
-		try {
-			runnable.run();
-		} catch (Exception e) {
-			throw new IllegalStateException("Cannot process mail " + toString(), e);
```

Comment:
Why did you move exception handling out? Now we have 3 places instead of 1 where we do the exception translation.

Reply:
I haven't moved it out - I just dropped this wrapping and now I'm letting exception to bubble up through the `MailboxProcessor#runMailboxLoop` callstack. It only needed to be wrapped into a runtime exception inside `yield` and `tryYield` methods to not pollute user's code.

Reply:
That's exactly what I meant by moving exception handling out. Now we have 3 places instead of 1 where we need to wrap it.

Reply:
Why 3? I see just `yield` and `tryYield`.

Besides that not wrapping the exception `MailboxProcessor#runMailboxLoop` is more or less whole point of this change/feature.

Reply:
Oh, I missed that, then it's even worse. The whole point of the exception translation is to actually use the the `#toString()` of Mail to give some context information about the mail being processed for failure debugging.

Reply:
This comment has not been addressed, but is important to me. It reverts some changes done in FLINK-14199 for improved error reporting.

Reply:
Ok, I see.

Btw, clearly this try/catch and `toString` was missing a test coverage